src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -2,23 +2,25 @@
 
 from datetime import datetime
 from pathlib import Path
+from typing import Optional
 
 from loguru import logger
 
 from dsp_tools.commands.xmlupload.models.deserialise.deserialise_value import XMLBitstream
 from dsp_tools.commands.xmlupload.models.deserialise.xmlresource import BitstreamInfo
 from dsp_tools.commands.xmlupload.models.deserialise.xmlresource import XMLResource
+from dsp_tools.commands.xmlupload.models.ingest import IngestClient
 from dsp_tools.commands.xmlupload.models.permission import Permissions
-from dsp_tools.commands.xmlupload.models.sipi import Sipi
 from dsp_tools.models.exceptions import PermanentConnectionError
 
 
 def handle_media_info(
     resource: XMLResource,
     media_previously_uploaded: bool,
-    sipi_server: Sipi,
+    ingest_client: IngestClient,
     imgdir: str,
     permissions_lookup: dict[str, Permissions],
+    shortcode: str,
```

Comment:
Why is the shortcode not a class variable of the `IngestClient`?

Reply:
So you can reuse the same instance for multiple uploads, passing in the `shortcode` for each ingest.

Reply:
The xmlupload CLI command is always only called for 1 project, so I'm not sure if this makes sense

Reply:
IMHO: I see the client as a service which itself should know about how to contact the api but nothing more. This client is able to interact with the dsp-ingest api and all parameters for a single call should be parameters on the respective function calling that api endpoint.  For me it is confusing to put the shortcode into the instance while the filename for the same api call/client funciton stays a parameter.

It would also be simpler to reuse the `IngestClient` later should that single project requirement change at some point. 

The only "drawback" is that the project code parameter must be available when calling the api which requires us to pass it down the call stack currently, which in most cases is done by the `UploadState` so the additional parameters are kept to minimum.